src/lib/protocols/credentials/CredentialService.ts

```diff
@@ -309,7 +314,13 @@ export class CredentialService extends EventEmitter {
     const prevState = credential.state;
     credential.state = newState;
     await this.credentialRepository.update(credential);
-    this.emit(EventType.StateChanged, { credential, prevState });
+
+    const event: CredentialStateChangedEvent = {
+      credential: credential,
```

Comment:
Couldn't we have just `credential` instead of `credential: credential`? 🤔 

```
const event: CredentialStateChangedEvent = {
      credential,
      prevState,
}
```

Reply:
```suggestion
      credentialRecord: credential,
```